core/src/test/scala/integration/kafka/api/PlaintextConsumerTest.scala

```diff
@@ -312,4 +314,47 @@ class PlaintextConsumerTest extends BaseConsumerTest {
     consumeAndVerifyRecords(this.consumers(0), 5, 5)
   }
 
+  @Test
+  def testFetchInvalidOffset() {
+    this.consumerConfig.setProperty(ConsumerConfig.AUTO_OFFSET_RESET_CONFIG, "none");
+    val consumer0 = new KafkaConsumer(this.consumerConfig, new ByteArrayDeserializer(), new ByteArrayDeserializer())
+
+    // produce one record
+    val totalRecords = 2
+    sendRecords(totalRecords, tp)
+    consumer0.assign(List(tp).asJava)
+
+    // poll should fail because there is no offset reset strategy set
+    intercept[NoOffsetForPartitionException] {
+      consumer0.poll(50)
+    }
+
+    // seek to out of range position
+    val outOfRangePos = totalRecords + 1
+    consumer0.seek(tp, outOfRangePos)
+    intercept[OffsetOutOfRangeException] {
```

Comment:
The OffsetOutOfRangeException includes a map with the partitions that were out of range and the offsets. Can you add an assertion which checks these values are correct?